frontend/src/views/LevelView.vue

```diff
@@ -41,56 +40,104 @@ import { ErrorMessage, SuccessMessage } from '@/services/messages'
 import { useRouter } from 'vue-router'
 import { HttpServerError } from '@/services/http'
 import { useLevelStore } from '@/stores/levels'
+import type { Role } from '@/domain/Role'
 
 const roleStore = useRolesStore()
-const role = ref<Role>()
-const newLevel = ref<Level>({
-  name: '',
-  minCoefficient: 0,
-  maxCoefficient: 0,
-  link: ''
-})
-const router = useRouter()
-const newLevelTitle = computed(() => {
-  return `${role.value?.name || ''} - ${newLevel.value.name || 'New Level'}`
-})
 const levelsStore = useLevelStore()
+const router = useRouter()
+const level = ref<Level>({} as Level)
+const role = ref<Role>()
+const isEditMode = ref(false)
+const roleId = ref()
+
+const localLevel = ref<Level>({ name: '', minCoefficient: 0, maxCoefficient: 0, link: '' })
+
+const dialogButtonLabel = computed(() => (isEditMode.value ? 'Save' : 'Create'))
+const dialogTitle = computed(() =>
+  isEditMode.value
+    ? role.value?.name + ' - Edit Level'
+    : role.value?.name +
+      (localLevel.value.name ? ' - ' + localLevel.value.name : ' - ' + ' New Level')
```

Comment:
This ternary is too complex. When you have a ternary too complex like this, you have to create a separated method to deal with this, in this case, the title